lib/src/main/java/com/hootsuite/nachos/NachoTextView.java

```diff
@@ -727,6 +727,128 @@ public void setTextWithChips(@Nullable List<ChipInfo> chips) {
         endUnwatchedTextChange();
     }
 
+    /**
+     *
+     * @return String of unchipped text that AutoCompleteTextView sees
+     */
+
+    public String getTextForSuggestion() {
+        if (getChipTokenizer() != null) {
+            int end = getSelectionEnd();
+            Editable editable = getText();
+            int start = getChipTokenizer().findTokenStart(editable, end);
+            if (start != -1 && end != -1) {
+                if (end < start) {
+                    start = end;
+                }
+                return editable.subSequence(start, end).toString();
+            }
+        }
+        return "";
+    }
+
+    /**
+     *
+     * @param text to check if exists with Chip
+     * @return true if an existing chip contains text and null object
+     */
+    public boolean isAlreadyAdded(CharSequence text) {
+        return isAlreadyAdded(text, null);
+    }
+
+    /**
+     *
+     * @param text to check if exists with Chip
+     * @param object to check if exists with Chip
+     * @return true if an existing chip contains text and object
+     */
+    public boolean isAlreadyAdded(CharSequence text, Object object) {
+        List<Chip> chips = getAllChips();
+        for (Chip chip: chips) {
+            CharSequence existingText = chip.getText();
+            Object existingObject = chip.getData();
+            if (text.equals(existingText) &&
+                    (object == null && existingObject == null) || (object != null && object.equals(existingObject))) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     *
+     * @param text creates a chip with specified text
+     */
+    public void insertChip(CharSequence text) {
+        insertChip(text, null);
+    }
+
+    /**
+     *
```

Comment:
Inserting seems pretty useful and this can stay in this PR. New public API methods should have better documentation to give users the contract that this method is agreeing to do. Please fill this out for this method and the previous `insertChip`